robot_calibration/src/optimizer.cpp

```diff
@@ -146,32 +144,42 @@ int Optimizer::optimize(OptimizationParams& params,
 
         // Create the block
         ceres::CostFunction * cost = Camera3dToArmError::Create(
-          dynamic_cast<Camera3dModel*>(models_[camera_name]),
-          models_[arm_name],
-          offsets_.get(), data[i]);
+            dynamic_cast<Camera3dModel*>(models_[camera_name]),
+            models_[arm_name],
+            offsets_.get(), data[i]);
+
+        int index = -1;
+        for (size_t k =0; k < data[i].observations.size() ; k++)
+        {
+            if ( data[i].observations[k].sensor_name == camera_name)
```

Comment:
this is still indented wrong, and should not have a space after opening parenthesis